extensions/amp-story/1.0/amp-story.js

```diff
@@ -1430,8 +1438,14 @@ export class AmpStory extends AMP.BaseElement {
               this.advancement_.getProgress()
             );
           }
+          storePageIndex = pageIndex;
         }
 
+        this.storeService_.dispatch(Action.CHANGE_PAGE, {
+          id: targetPageId,
+          index: storePageIndex,
+        });
```

Comment:
Just making sure: can moving this have any side effect?

Reply:
After doing a sanity check it doesn't seem to.